Spring annotations for dependency injection added

MartaKwiatek opened the pull request.

MartaKwiatek committed:
Spring annotations for dependency injection added

andrzeju commented on app/src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java:
```diff
@@ -43,8 +44,8 @@ public int save(Invoice invoice) {
         idService.setId();
```
nazwa setId() może być trochę myląca - nie wiadomo na co ustawiamy wartość. Zmieniłbym nazwę na coś co mówi o zachowaniu metody (szczególnie o tym, że robimy tam inkrementację, bo to nie jest oczywiste jak się patrzy na ten serwis)

andrzeju reviewed (changes requested):
oprócz komentarza to jeszcze napisałem na chacie

MartaKwiatek committed:
renamed method setId to incrementId (in IdService)
deleted unnecessary if statement, when use method createNewFile (in IdService and FileBasedDatabase)
